src/main/java/net/snowflake/ingest/connection/TelemetryService.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2022 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.ingest.connection;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Snapshot;
+import com.codahale.metrics.Timer;
+import net.snowflake.client.jdbc.internal.apache.http.impl.client.CloseableHttpClient;
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.ObjectMapper;
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.node.ObjectNode;
+import net.snowflake.client.jdbc.telemetry.TelemetryClient;
+import net.snowflake.client.jdbc.telemetry.TelemetryUtil;
+import net.snowflake.ingest.utils.Logging;
+
+/**
+ * Telemetry service to collect logs in the SDK and send them to Snowflake through the JDBC client
+ * telemetry API
+ */
+public class TelemetryService {
+  // Enum for different client telemetries
+  enum TelemetryType {
+    STREAMING_INGEST_LATENCY_IN_SEC("streaming_ingest_latency_in_sec"),
+    STREAMING_INGEST_CLIENT_FAILURE("streaming_ingest_client_failure"),
```

Comment:
Should we add `_COUNT` or something here?

Reply:
this is not an aggregate counter, we will have one per failure and it will contain the actual failure exception